core/src/stores/postgres.rs

```diff
@@ -1683,7 +1706,9 @@ impl Store for PostgresStore {
                 let stmt = c
                     .prepare(
                         format!(
-                            "SELECT COUNT(*) FROM data_sources_documents WHERE {}",
+                            "SELECT COUNT(*) FROM data_sources_documents dsd \
+                               INNER JOIN data_sources_nodes dsn ON dsn.document=dsd.id \
```

Comment:
I see why we need that JOIN for the count to work (dsn.parents vs dsd.tags_array and dsd.timestamp)

I'm worried this will be too expensive? Any chance we can test at scale vs current state of affairs?

Reply:
Thanks for the detailed analysis. Let's keep a close eye on it :+1:

Reply:
I also think we could make the count result optional, as it's just ignore in many queries ( for example the `retrieve_chunks_without_query` just ignores it )